### ISSUE TITLE
Reflow cursor position when terminal window is resized

### ISSUE THREAD

**Issue by Anomalocaridid** [wez/wezterm#1821]
**Is your feature request related to a problem? Please describe.**
Due to an issue in the shell I use, zsh, when I resize a terminal window, the text gets messed up when it gets reflowed. There is an unofficial patch that I can compile zsh with that fixes it, but it requires that the terminal emulator I use reflows the cursor position as well when the window gets resized.

**Describe the solution you'd like**
I would like Wezterm to reflow the cursor position along with the text when the terminal window is resized, either as default behavior or as an option that can be configured.

**Describe alternatives you've considered**
I am not sure what other options I have other than continuing to live with it.

**Additional context**
Here is information on the zsh patch that I was using in case that would be relevant: https://github.com/romkatv/powerlevel10k#zsh-patch

**Comment by wez** [comment]
wezterm does adjust the cursor position when the window is resized:

https://github.com/wez/wezterm/blob/8a0072add1aef18d607a404b7dba038d4ba8bec9/term/src/screen.rs#L274-L279

However, on Windows when ConPTY is in use, ConPTY tends to mangle the cursor position.

Which OS are you using?

**Comment by Anomalocaridid** [comment]
I use Linux. More specifically, the Sway edition of EndeavourOS. In that case, I think it is more likely that there may be some issue with either the patch that I used or how I applied it rather than an issue with WezTerm. It looks like I have the same issue with other terminal emulators. I probably should have tested it with another terminal emulator before making this issue. Thanks for your help.
